### ISSUE TITLE
Editing certificates doesn't work in the GUI

### ISSUE THREAD

**Issue by SamMousa** [rancher/rancher#8674]
**Rancher Versions:**
Server: 1.5.6

I'm aware that the API (and by extension the GUI) does not ever return the private key for a certificate.

**Steps to Reproduce:**
Try to change the name of a certificate.

**Results:** 
Error message indicating that key is required.

While doing this via the API I discovered that simply omitting the `key` in the data allows me to rename a certificate.

This is weird, from a REST perspective submitting data should replace the resource. Instead it seems to do some patching on the resource instead. (Note that this is documented in detail [here](https://github.com/rancher/api-spec/blob/master/specification.md#update-operation)).

This leads me to the conclusion that there is a difference between omitting a key and specifying a null value. 
````
This works
{
    "id": "1c3",
    "name": "newname",
...
}

This doesn't work
{
    "id": "1c3",
    "name": "newname",
    "key": null
...
}
````

All of this makes sense, but it leaves a problem for GUI usage.
The GUI tells me fields required on update:
![image](https://cloud.githubusercontent.com/assets/547021/25742250/ef40cdb8-318e-11e7-9596-e44b2143225b.png)

In reality however these attributes are required on creation only, and I should be allowed to update just one of them. The client side GUI code should then detect which fields I have changed and only include those in the submission.

Currently change a certificate name without providing the private key again does not work. Having the private key duplicated and always at hand doesn't seem realistic to me.

tldr; The GUI should only submit dirty (altered) inputs to the API.

**Comment by westlywright** [comment]
Should be fixed when the following commit is merged
ccb2ba85f342a9c177e118283f95911c2a9e1cdb

**Comment by tfiduccia** [comment]
Version - master 2.0 7/31
Verified fixed
